DependencyInjection/CmfResourceExtension.php

```diff
@@ -34,64 +35,120 @@ public function load(array $configs, ContainerBuilder $container)
 
         $config = $processor->processConfiguration($configuration, $configs);
 
-        $this->loadRepositories($config['repository'], $container);
+        $this->loadRepositories($config['repositories'], $container);
     }
 
-    private function loadRepositories($config, $container)
+    private function loadRepositories(array $repositories, ContainerBuilder $container)
     {
-        foreach (array(
+        $methods = array(
             'composite' => 'createCompositeRepository',
             'doctrine_phpcr' => 'createDoctrinePhpcrRepository',
             'doctrine_phpcr_odm' => 'createDoctrinePhpcrOdmRepository',
             'filesystem' => 'createFilesystemRepository',
-        ) as $type => $createMethod) {
-            foreach ($config[$type] as $repoAlias => $repoConfig) {
-                $definition = $this->$createMethod($repoConfig);
-                $definition->addTag('cmf_resource.repository', array('type' => $type, 'alias' => $repoAlias));
-                $container->setDefinition('cmf_resource.repository.' . $type . '.' . $repoAlias, $definition);
+        );
+       
+        foreach ($repositories as $alias => $repository) {
+            if (!isset($methods[$repository['type']])) {
+                throw new InvalidConfigurationException(sprintf(
+                    'Unexpected repository type "%s", known types: %s',
+                    $repository['type'],
+                    implode(', ', array_keys($methods))
+                ));
             }
+
+            $definition = $this->{$methods[$repository['type']]}($repository['options'], $alias);
+            $definition->addTag('cmf_resource.repository', array('type' => $repository['type'], 'alias' => $alias));
+            $container->setDefinition('cmf_resource.repository.' . $repository['type'] . '.' . $alias, $definition);
```

Comment:
as repository names are now globally unique, would it make sense to still prefix the service ID with the repository type?